### PR TITLE
feat: add omnifunc for query filetype

### PR DESCRIPTION
**Pull request by theHamsta**
A simple omnifunc for the query filetype.
noidea

### CODE REVIEW

File: autoload/nvim_treesitter.vim [hunk 1]
```diff
@@ -21,3 +21,17 @@ endfunction
 function! nvim_treesitter#indent() abort
 	return luaeval(printf('require"nvim-treesitter.indent".get_indent(%d)', v:lnum))
 endfunction
+
+function! nvim_treesitter#query_omnifunc(findstart, base) abort
+  if a:findstart
+    " locate the start of the word
+    let line = getline('.')
+    let start = col('.') - 1
+    while start > 0 && ((line[start - 1] =~ '\a' || line[start - 1] =~ '-' || line[start - 1] =~ '"') && line[start - 1] !~ '(' )
```
Comment (theHamsta) [inline autoload/nvim_treesitter.vim:30]: should probably be a `strmatchpos`

File: autoload/nvim_treesitter.vim [hunk 2]
```diff
@@ -21,3 +21,14 @@ endfunction
 function! nvim_treesitter#indent() abort
 	return luaeval(printf('require"nvim-treesitter.indent".get_indent(%d)', v:lnum))
 endfunction
+
+function! nvim_treesitter#query_omnifunc(findstart, base) abort
+  if a:findstart
+    " locate the start of the word
+    let start = col('.') - 2
```
Comment (theHamsta) [inline autoload/nvim_treesitter.vim:28]: -2. Why?

Comment (theHamsta) [reply autoload/nvim_treesitter.vim:28]: Ok, without the current cursor position and transform into 0-based indexing

File: autoload/nvim_treesitter.vim [hunk 3]
```diff
@@ -21,3 +21,14 @@ endfunction
...
 endfunction
+
+function! nvim_treesitter#query_omnifunc(findstart, base) abort
+  if a:findstart
+    " locate the start of the word
+    let start = col('.') - 2 " -2 necessary to exclude current cursor position and to transfer into 0-based indexing
+    let result = matchstrpos(getline('.')[0:start], '\v("|\\|\-|\w)*$')
+    return result[1]
+  else
+    return luaeval(printf("require'nvim-treesitter.query_filetype'.omnifunc([[%s]])", a:base))
```
Comment (theHamsta) [inline autoload/nvim_treesitter.vim:32]: can this be done in a way that will not break when you enter `]]`?

**Comment by vigoux** [comment]
I did some things in `architext.nvim` that I actually never ported, mayde that'll help a bit.

**Comment by theHamsta** [comment]
@vigoux then, we should use your code! Maybe you can take the `vim.fn.escape` part of my PR to escape backslashes?

**Comment by vigoux** [comment]
Yup, that looks like a great idea.
I am starting to think that we should host all the query-related things into a custom plugin, which we could then reuse.

**Comment by theHamsta** [comment]
We are also missing a `:TSEditQuery` which opens the appropriate query file and prompts user to select if there are multiple.

**Comment by theHamsta** [comment]
Should we leave this open until we have a proper solution? Maybe move this to playground? I think that this feature is so minor that most users won't bother to install a plugin for it but are just happy if it just works.

**Comment by vigoux** [comment]
Maybe moving that to the playground is better, this makes the plugin more like a dev-tools plugin for nvim-treesitter, and we can point to that in the contributing and things.

**Pull request closed without merging**